### DISCUSSION TITLE
Miryoku switching Spacebar and Enter around.

### DISCUSSION THREAD

**Discussion by Supgeekhack** [manna-harbour/miryoku#189] [Q&A] [answered]
Hello.

I love the miryoku layout but, i am right handed for the spacebar. Is there a way to change it around? I figured around how to enable the 6th column but, i can't seem to switch the enter and spacebar without the keyboard spuwing gibberish when switching 34 and 35
around.  This is for CRKBD
![image](https://user-images.githubusercontent.com/38331135/219494806-38ca0219-424e-491c-8b7f-b286eb9a4933.png)

**Comment by manna-harbour** [comment, marked as answer]
To map extra keys, see the Mapping section at https://github.com/manna-harbour/miryoku/discussions/85. You can also just edit `config.h` as you've done, but using `custom_config.h` keeps your changes separate.

To move keys around within layers, see the Substitute Layer section at https://github.com/manna-harbour/miryoku/discussions/85. If you instead move keys in the mapping the swap will apply to all layers. You could change the mapping per layer as described above, but you'll have to do it for each layout, whereas with a substituted layer it will apply to all layouts.

The only built-in option for right Space is to use the flip layers option. See https://github.com/manna-harbour/miryoku/tree/master/docs/reference#flip. If you want right Space without flip, you probably don't want to swap Enter and Space directly as you'll want Space on a home thumb. Instead you could rotate Enter, Space, and Backspace so that Enter is on left home thumb, Backspace is on right secondary thumb, and Space is on right home thumb. You may also want to make the same change on Nav for consistency, or swap Backspace and Enter on Nav for Backspace consistency while still including Enter for editing.

Also consider Space on both home thumbs, Backspace on right secondary thumb on Base and Nav, and Enter on right home on Nav only, while you transition to left Space, then later switch to default. Adapting to Space on the other side can take a lot of effort, but swapping Enter and Backspace later should be much easier.

**Reply by Supgeekhack** [reply]
Ah that makes sense. My English isn't the best so i may interpret some stuff wrong in the documentation. But, do you rename config.h to custom_config.h or do you make a copy of the config.h in the same directory and rename it to custom_config.h?

In my case i just define the baselayer QWERTY in the custom_config.h and edit to my liking. Do i need to add parameters when compiling or does it automatically pick up the custom_config.h? 

make clean crkbd:manna-harbour_miryoku \ MIRYOKU_ALPHAS=QWERTY \ would suffice?  

It's probably simpel to do if you understand how it works but, the steps of editing a substitute layer confuses me. Thanks for your elaborate explanation though it did help me get further and will look further into it.

**Reply by manna-harbour** [reply]
To substitute a custom Base with Space on both sides, starting from QWERTY, on Miryoku QMK, add the following to https://github.com/manna-harbour/miryoku_qmk/blob/miryoku/users/manna-harbour_miryoku/custom_config.h.

```
#define  MIRYOKU_LAYER_BASE \
KC_Q,              KC_W,              KC_E,              KC_R,              KC_T,              KC_Y,              KC_U,              KC_I,              KC_O,              KC_P,              \
LGUI_T(KC_A),      LALT_T(KC_S),      LCTL_T(KC_D),      LSFT_T(KC_F),      KC_G,              KC_H,              LSFT_T(KC_J),      LCTL_T(KC_K),      LALT_T(KC_L),      LGUI_T(KC_QUOT),   \
LT(U_BUTTON,KC_Z), ALGR_T(KC_X),      KC_C,              KC_V,              KC_B,              KC_N,              KC_M,              KC_COMM,           ALGR_T(KC_DOT),    LT(U_BUTTON,KC_SLSH),\
U_NP,              U_NP,              LT(U_MEDIA,KC_ESC),LT(U_NAV,KC_SPC),  LT(U_MOUSE,KC_TAB),LT(U_SYM,KC_BSPC),  LT(U_NUM,KC_SPC), LT(U_FUN,KC_DEL),  U_NP,              U_NP

```

You can edit to change the thumb keys, and also substitute Nav in the same way.

The location of `custom_config.h` is mentioned in the **custom_config.h** section in https://github.com/manna-harbour/miryoku/discussions/85.

The layer data you're starting from is `MIRYOKU_ALTERNATIVES_BASE_QWERTY` defined at https://github.com/manna-harbour/miryoku_qmk/blob/d92f5d177210891b34c9207a6aa19b900d50439c/users/manna-harbour_miryoku/miryoku_babel/miryoku_layer_alternatives.h#L119-L123.

The macro name, `MIRYOKU_LAYER_BASE`, is from https://github.com/manna-harbour/miryoku_qmk/blob/d92f5d177210891b34c9207a6aa19b900d50439c/users/manna-harbour_miryoku/miryoku_babel/miryoku_layer_selection.h#L11

`custom_config.h` is enabled automatically.

Layer substitution for Base will override any `MIRYOKU_ALPHAS` option, so there's no need to specify that.
